app/src/main/java/com/example/androidtrainingproject/login/LoginScreen.kt

```diff
@@ -47,69 +62,97 @@ fun LoginScreen(navigateToProductDetails: () -> Unit = {}) {
+            value = usernameField,
+            label = { Text(text = stringResource(id = R.string.label_email)) },
+            colors = TextFieldDefaults.colors(
+                errorIndicatorColor = ErrorDarkRed,
+                errorContainerColor = Color.Transparent,
+                unfocusedContainerColor = Color.Transparent,
+                focusedContainerColor = Color.Transparent,
+            ),
+            isError = markInputsAsWrong,
+            onValueChange = { newValue -> usernameField = newValue }
         )
         OutlinedTextField(
-            modifier = Modifier
-                .padding(bottom = 10.dp)
-                .fillMaxWidth(),
-            value = passwordField.value,
-            label = { Text(text = "Password")},
-            onValueChange = {
-                    newValue -> passwordField.value = newValue
-            },
-            visualTransformation = if (isPasswordVisible.value) VisualTransformation.None else PasswordVisualTransformation(),
+            modifier = Modifier.fillMaxWidth(),
+            value = passwordField,
+            label = { Text(text = stringResource(id = R.string.label_password)) },
+            colors = TextFieldDefaults.colors(
+                errorIndicatorColor = ErrorDarkRed,
+                errorContainerColor = Color.Transparent,
+                unfocusedContainerColor = Color.Transparent,
+                focusedContainerColor = Color.Transparent,
+            ),
+            isError = markInputsAsWrong,
+            onValueChange = { newValue -> passwordField = newValue },
+            visualTransformation = if (isPasswordVisible) VisualTransformation.None else PasswordVisualTransformation(),
             trailingIcon = {
                 IconButton(onClick = {
-                    isPasswordVisible.value = !isPasswordVisible.value
+                    isPasswordVisible = !isPasswordVisible
                 }) {
                     Image(
                         modifier = Modifier
-                            .height(36.dp)
-                            .width(32.dp),
+                            .height(dimensionResource(id = R.dimen.icon_normal))
+                            .width(dimensionResource(id = R.dimen.icon_normal)),
                         painter = painterResource(id = passwordIcon),
-                        contentDescription = "toggle password visibility"
+                        contentDescription = stringResource(id = R.string.toggle_password_visibility)
                     )
                 }
             },
         )
 
         if (loginViewModel.loginFailure) {
-            Toast.makeText(LocalContext.current, "Invalid login credentials", Toast.LENGTH_SHORT).show()
+            Toast.makeText(LocalContext.current, stringResource(id = R.string.login_failure), Toast.LENGTH_SHORT).show()
+            markInputsAsWrong = true
             loginViewModel.loginFailure = false
         }
 
         Button(
```

Comment:
Not sure if you're going to reuse this button with those very styles but it's generally a good idea to export it as a shared reusable composable.